splat.pl

```diff
@@ -0,0 +1,953 @@
+##***************************************************************************
+    for $slot (0, 2, 4, 6) {
+	if (defined $cal{IMGCOL0}[$slot]) { 
+	    $col0 = $cal{IMGCOL0}[$slot];
+	    $row0 = $cal{IMGROW0}[$slot];
+	} else {$col0 = 0; $row0 = 0};
+	$out .= join '',"-" x 11, "SLOT ",$slot;
+	$out .= sprintf (" (%4.0f,%4.0f)", $col0,$row0);
+	$out .= sprintf (" (%7.0f %3s)", $img_total[$slot], $opt_scale ? 'DN' : 'raw');
+	$out .= join '', "-"x18, "SLOT ",$slot+1;
+	if (defined $cal{IMGCOL0}[$slot+1]) { 
+	    $col0 = $cal{IMGCOL0}[$slot+1];
+	    $row0 = $cal{IMGROW0}[$slot+1];
+	} else {$col0 = 0; $row0 = 0};
+	$out .= sprintf (" (%4.0f,%4.0f)", $col0,$row0);
+	$out .= sprintf (" (%7.0f %3s)", $img_total[$slot+1], $opt_scale ? 'DN' : 'raw');
+	$out .= join '', "-"x8,"\n";
+	for $i (7, 6, 5, 4, 3, 2, 1, 0) {
+	    $out .= join '', "| ";
+	    for $j (0 .. 7) {
+		$out .= sprintf ("%5.0f ", $img_array[$slot][$i][$j]);
+	    }
+	    $out .= join '', "| ";
+	    for $j (0 .. 7) {
+		$out .= sprintf ("%5.0f ", $img_array[$slot+1][$i][$j]);
+	    }
+	    $out .= join '', "|\n";
+	}
+    }
+    $out .= join '', "-" x 101, "\n";
+}
+
+##***************************************************************************
+sub decom_aca_packet {
+##***************************************************************************
+    my $slot;
+    my $img_type;
+    my $offset;
+
+    # Extract global status stuff
+    $startbit = 0;
+    $integ_time = extract ($aca_packet_vec, 16) * 0.016;
+
+    $startbit = 16;
+    $highbgd = extract ($aca_packet_vec, 1);
+    $ramfail = extract ($aca_packet_vec, 1);
+    $romfail = extract ($aca_packet_vec, 1);
+    $pwrfail = extract ($aca_packet_vec, 1);
+    $calfail = extract ($aca_packet_vec, 1);
+    $chkfail = extract ($aca_packet_vec, 1);
+    $startbit = 23;
+    $synfail = extract ($aca_packet_vec, 1);
+    $cmd_cnt = extract($aca_packet_vec, 6);
+    extract ($aca_packet_vec, 2);
+    $cmd_prog= extract ($aca_packet_vec, 6);
+
+    # If the PEA is off, then clear each the image slot, redrawing if needed
+
+    if (substr ($aca_packet, 3, 40) eq "\377"x40) {  # PEA off
+        $pea_on = 0;
+	for $slot (0..7) {
+	    my $update_slot = 0;
+	    for $i (0..7) {
+		for $j (0..7) {
+		    $update_slot = 1 if ($img_array[$slot][$i][$j] != 0.0);
+		    $img_array[$slot][$i][$j] = 0.0;
+		}
+	    }
+	    draw_slot($slot);
+	}
+	return;
+    }
+
+    $pea_on = 1;
+    $startbit = 40;		# start of "image types" in aca_packet_bits
+    for $slot (0 .. 7) {
+#       next unless (extr_val ($aca_packet_vec, 68+$slot*216, 2));    # idle slot?
+	$img_type = extract ($aca_packet_vec, 3);
+	$cal{IMGTYPE}[$slot] = $img_type;
+
+	# Put header data into $img_data[$slot] if needed
+	if ($img_type == 0 || $img_type == 1 || $img_type == 4) {
+	    $img_data[$slot] = substr $aca_packet, 0, 5;
+	}
+
+	# identify a 6x6 or 8x8 readout slot, for printing out temperatures
+	if ($img_type == 2 || $img_type == 5) {$slot68 = $slot};
```

Comment:
The hdr3 telem would be great, but the hdr3_decom.pl script seems to basically be splat.pl and doesn't seem to have the hires temperatures handy.  I might just be missing it.